src/change_dump_date.py

```diff
@@ -1,47 +1,171 @@
-# xconv less confusing.
+def year_value(value):
+    """
+    Ensure the year is a positive integer.
+    """
+    ivalue = int(value)
+    if ivalue < 1:
+        raise argparse.ArgumentTypeError(f"Invalid year: {value}. Must be a positive integer.")
+    return ivalue
 
-import getopt, sys
-import umfile
-from um_fileheaders import *
+def month_value(value):
+    """
+    Ensure the month is between 1 and 12 (inclusive).
+    """
+    ivalue = int(value)
+    if ivalue < 1:
+        raise argparse.ArgumentTypeError(f"Invalid month: {value}. Must be a positive integer.")
+    if ivalue > 12:
+        raise argparse.ArgumentTypeError(f"Invalid month: {value}. Must be between 1 and 12.")
+    return ivalue
 
-ifile = sys.argv[1]
+def day_value(value):
+    """
+    Ensure the day is between 1 and 31 (inclusive).
+    """
+    ivalue = int(value)
+    if ivalue < 1:
+        raise argparse.ArgumentTypeError(f"Invalid day: {value}. Must be a positive integer.")
+    if ivalue > 31:
+        raise argparse.ArgumentTypeError(f"Invalid day: {value}. Must be between 1 and 31.")
+    return ivalue
 
-f = umfile.UMFile(ifile, "r+")
 
-print("Initial Time", f.fixhd[FH_DTYear], f.fixhd[FH_DTMonth],
-      f.fixhd[FH_DTDay], f.fixhd[FH_DTHour], f.fixhd[FH_DTMinute],
-      f.fixhd[FH_DTSecond])
+def parse_args():
+    """
+    Parse the command line arguments.
 
-print("Valid Time", f.fixhd[FH_VTYear], f.fixhd[FH_VTMonth],
-      f.fixhd[FH_VTDay], f.fixhd[FH_VTHour], f.fixhd[FH_VTMinute],
-      f.fixhd[FH_VTSecond])
+    Parameters
+    ----------
+    None
 
-s = input('Enter year month day\n')
-s = s.split()
-year = int(s[0])
-month = int(s[1])
-day = int(s[2])
+    Returns
+    ----------
+    args_parsed : argparse.Namespace
+        Argparse namespace containing the parsed command line arguments.
+    """
+    parser = argparse.ArgumentParser(description="Modify UM dump file timestamps.")
+    parser.add_argument('ifile', metavar="INPUT_PATH", help='Path to the input file.')
+    parser.add_argument('-y', '--year', type=year_value, help='New year value.', required=True)
+    parser.add_argument('-m', '--month', type=month_value, help='New month value (1-12).', required=True)
+    parser.add_argument('-d', '--day', type=day_value, help='New day value (1-31).', required=True)
+    parser.add_argument('-o', '--output', dest='output_path', metavar="OUTPUT_PATH",
+                        help='Path to the output file. If omitted, the default output file is created by appending "_perturbed" to the input path.')
+    parser.add_argument('--validate', action='store_true',
+        help='Validate the output fields file using mule validation.')
 
-print("Using", year, month, day)
+    return parser.parse_args()
 
-f.fixhd[FH_DTYear] = f.fixhd[FH_VTYear] = year
-f.fixhd[FH_DTMonth] = f.fixhd[FH_VTMonth] = month
-f.fixhd[FH_DTDay] = f.fixhd[FH_VTDay] = day
+def change_fileheader_date(ff, new_year, new_month, new_day):
```

Comment:
```suggestion
def change_header_date_file(ff, new_year, new_month, new_day):
```
It is very minor, but I initially thought that this function and `change_fieldheader_date` had the same name. I had to look for a bit to get the difference.
I think putting `file` and `field` at the end might help catching that difference more quickly.

Reply:
I think this is resovled